action/protocol/rewarding/fund.go

```diff
@@ -146,3 +219,29 @@ func DepositGas(ctx context.Context, sm protocol.StateManager, amount *big.Int)
 	}
 	return rp.Deposit(ctx, sm, amount, iotextypes.TransactionLogType_GAS_FEE)
 }
+
+// DepositGasAndBurn deposits gas into the rewarding fund and burns the base fee
+func DepositGasAndBurn(ctx context.Context, sm protocol.StateManager, amount, baseFee *big.Int) ([]*action.TransactionLog, error) {
+	// If the gas fee is 0, return immediately
+	if amount.Cmp(big.NewInt(0)) == 0 {
+		return nil, nil
+	}
+	blkCtx := protocol.MustGetBlockCtx(ctx)
+	if blkCtx.BlockHeight == 0 {
+		return nil, nil
+	}
+	reg, ok := protocol.GetRegistry(ctx)
+	if !ok {
+		return nil, nil
+	}
+	rp := FindProtocol(reg)
+	if rp == nil {
+		return nil, nil
+	}
+	burn, _ := address.FromString(address.ZeroAddress)
+	return rp.depositAndBurn(ctx, sm, burn, amount, baseFee, iotextypes.TransactionLogType_GAS_FEE, iotextypes.TransactionLogType_NATIVE_TRANSFER)
```

Comment:
Is it appropriate to use TransactionLogType_NATIVE_TRANSFER to denote burning? Will it cause any misunderstandings (e.g. transfer basefee to other people)?

Reply:
in blockchain, burn = send to the zero address
we can discuss, maybe send the amount to a struct (like `fund`) in rewarding protocol to keep track of all burned fees. and maybe can use this part for inflation